### PR TITLE
Remove Fluent NorthStar

### PR DESCRIPTION
**Pull request by emlynmac**
# What
Remove deprecated fluent N* library and replace with fluent 9 controls.

# Why
NorthStar is not fully compatible with React 18

# How Tested
CI, storybook

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/communication-react/review/beta/communication-react.api.md [hunk 1]
```diff
@@ -91,7 +92,8 @@ import { SyntheticEvent } from 'react';
 import { TeamsCall } from '@azure/communication-calling';
 import { TeamsCallAgent } from '@azure/communication-calling';
 import { TeamsMeetingLinkLocator } from '@azure/communication-calling';
-import { Theme } from '@fluentui/react';
+import { Theme as Theme_2 } from '@fluentui/react';
```
Comment (vhuseinova-msft) [inline packages/communication-react/review/beta/communication-react.api.md:95]: I'm wondering if it's possible to improve the types in this file?

File: packages/react-components/package.json [hunk 2]
```diff
@@ -35,12 +35,15 @@
   },
   "dependencies": {
     "@fluentui/react": "~8.110.4",
+    "@fluentui/react-components": "~9.27.4",
     "@fluentui/react-file-type-icons": "8.8.13",
     "@fluentui/react-hooks": "~8.6.12",
     "@fluentui/react-icons": "~2.0.194",
+    "@fluentui/react-migration-v0-v9": "~9.0.17",
```
Comment (vhuseinova-msft) [inline packages/react-components/package.json:42]: Do we still need these 2 dependencies?

File: packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx [hunk 3]
```diff
@@ -293,39 +308,119 @@ const MessageBubble = (props: ChatMessageComponentAsMessageBubbleProps): JSX.Ele
     handleOnInlineImageClicked
   ]);
 
+  const attached = message.attached === 'top' ? 'top' : message.attached === false ? 'top' : 'center';
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx:311]: can be updated to
```suggestion
  const attached = message.attached === 'top' || message.attached === false ? 'top' : 'center';
```

File: packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx [hunk 4]
```diff
@@ -293,39 +308,119 @@ const MessageBubble = (props: ChatMessageComponentAsMessageBubbleProps): JSX.Ele
...
+                messageContainerClasses?.body
+              )
+            }}
+            data-ui-id="chat-composite-message"
+            onTouchStart={() => setWasInteractionByTouch(true)}
+            onPointerDown={() => setWasInteractionByTouch(false)}
+            onKeyDown={() => setWasInteractionByTouch(false)}
+            onBlur={() => setWasInteractionByTouch(false)}
+            onClick={() => {
+              if (!wasInteractionByTouch) {
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx:402]: `OnClick` can be a separate func to remove duplications

File: packages/react-components/src/components/styles/MessageThread.styles.ts [hunk 5]
```diff
@@ -118,6 +163,46 @@ export const FailedMyChatMessageContainer: ComponentSlotStyle = {
...
+  body: {
+    ...defaultChatItemMessageContainer,
+    backgroundColor: 'rgb(199, 224, 244)'
+  }
+});
+
+/**
+ * @private
+ */
+export const useChatMyMessageClasses: (messageState?: MessageStatus) => Record<'body', string> = (
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/styles/MessageThread.styles.ts:179]: As `useChatMyMessageClasses` and `useChatMessageClasses` do the same, should we reuse code instead?
same for `useChatFailedMyMessageClasses` and `useChatFailedMessageClasses`, `chatNormalMyMessageClasses` and `chatNormalMessageClasses`

**Comment by vhuseinova-msft** [comment]
Not sure if it's expected

| Was | Now |
|------|------|
| image | image |

**Comment by vhuseinova-msft** [comment]
It looks like delete button doesn't work in context menu for a message in the MessageThread

**Comment by vhuseinova-msft** [comment]
Message with a Mention is broken
image

File: packages/communication-react/review/beta/communication-react.api.md (hunk 1 above)
Comment (emlynmac) [reply packages/communication-react/review/beta/communication-react.api.md:95]: It's auto-generated, so maybe in the future. There are lots of people on the api-extractor github asking questions about this.

File: packages/react-components/src/components/styles/MessageThread.styles.ts (hunk 5 above)
Comment (emlynmac) [reply packages/react-components/src/components/styles/MessageThread.styles.ts:179]: I moved the original logic that was in each of these out to the theme; these do indeed need to be refactored as a result.

**Pull request closed without merging**